src/engine/source/router/src/environmentManager.cpp

```diff
@@ -32,36 +32,44 @@ std::optional<base::Error> EnvironmentManager::addEnvironment(const std::string&
                                         name)};
     }
 
-    std::unique_lock<std::shared_mutex> lock(m_mutex);
-    if (m_environments.find(name) != m_environments.end())
+    // Create the environment
+    std::vector<RuntimeEnvironment> envs = {};
+    envs.reserve(m_numInstances);
+    for (std::size_t i = 0; i < m_numInstances; ++i)
     {
-        return base::Error {fmt::format("Environment '{}' already exists", name)};
+        auto env = RuntimeEnvironment {name};
+        const auto err = env.build(m_builder);
+        if (err)
+        {
+            return base::Error {err.value()};
+        }
+        envs.push_back(env);
     }
-
-    auto env = std::make_shared<RuntimeEnvironment>(name, m_numThreads, m_queue);
-    const auto err = env->build(m_builder);
-    if (err)
+    // Add the environment to the runtime list
     {
-        return base::Error {err.value()};
+        std::unique_lock<std::shared_mutex> lock(m_mutex);
+        if (m_environments.find(name) != m_environments.end())
+        {
+            return base::Error {fmt::format("Environment '{}' already exists.", name)};
+        }
+        m_environments.insert({name, std::move(envs)});
     }
-    m_environments.emplace(name, std::move(env));
 
     return std::nullopt;
 }
 
-std::optional<base::Error> EnvironmentManager::delEnvironment(const std::string& name)
+std::optional<base::Error> EnvironmentManager::deleteEnvironment(const std::string& name)
 {
     std::unique_lock<std::shared_mutex> lock(m_mutex);
     auto it = m_environments.find(name);
     if (it == m_environments.end())
     {
-        return base::Error {fmt::format("Environment '{}' does not exist", name)};
+        return base::Error {fmt::format("Environment '{}' doesn't exist.", name)};
```

Comment:
Revert this please, most of the messages are using "does not" instead of "doesn't". also the "." is supposed to be printed on a higher level

Reply:
done